calcite-adapter/src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java

```diff
@@ -333,24 +333,30 @@ private static Operand getMongoAggregateForAndOperator(final RexCall call, final
             sb.deleteCharAt(sb.length() - 1);
             sb.append("]}, false, null]}]}");
 
-            return new Operand(sb.toString());
+            return new Operand(
+                    sb.toString(),
+                    SimpleMatchTranslator.getAndOrOperator(operands, s),
+                    false);
         }
 
         private static Operand getMongoAggregateForOrOperator(final RexCall call, final List<Operand> operands, final String s) {
             final StringBuilder sb = new StringBuilder();
             sb.append("{$cond: [{$or: [");
             for (Operand value: operands) {
-                sb.append("{$eq: [true, ").append(value.getValue()).append("]},");
+                sb.append("{$eq: [true, ").append(value.getAggregationValue()).append("]},");
             }
             sb.deleteCharAt(sb.length() - 1);
             sb.append("]}, true,");
             sb.append("{$cond: [{$and: [");
             for (Operand value: operands) {
-                sb.append("{$eq: [false, ").append(value.getValue()).append("]},");
+                sb.append("{$eq: [false, ").append(value.getAggregationValue()).append("]},");
             }
             sb.deleteCharAt(sb.length() - 1);
             sb.append("]}, false, null]}]}");
-            return new Operand(sb.toString());
+            return new Operand(
```

Comment:
Can support OR in simple match BUT this can only be nested in other ANDs and ORs not in a comparison operator so isQuerySyntax is false here